src/status_im/extensions/ethereum.cljs

```diff
@@ -116,16 +118,24 @@
 
 (handlers/register-handler-fx
  :extensions/ethereum-logs
- (fn [_ [_ _ {:keys [fromBlock toBlock address topics blockhash on-result]}]]
-   (let [parsed-topics (mapv parse-topic topics)
-         args {:jsonrpc "2.0"
-               :method constants/web3-get-logs
-               :params  [{:fromBlock (ensure-hex-bn fromBlock)
-                          :toBlock   (ensure-hex-bn toBlock)
-                          :address   address
-                          :topics    parsed-topics
-                          :blockhash blockhash}]}
-         payload (types/clj->json args)]
-     (status/call-private-rpc payload #(let [{:keys [error result]} (types/json->clj %1)
-                                             response (merge {:result result} (when error {:error error}))]
-                                         (re-frame/dispatch (on-result response)))))))
\ No newline at end of file
+ (fn [{db :db} [_ _ {:keys [fromBlock toBlock address topics blockhash on-result] :as arguments}]]
+   (if (ens/is-valid-eth-name? address)
+     (let [{:keys [web3 network]} db
+           network-info (get-in db [:account/account :networks network])
+           chain (ethereum/network->chain-keyword network-info)
+           registry (get ens/ens-registries chain)]
+       (do
+         (print "network: " network "\nchain: " chain)
+         (ens/resolver web3 registry address #(re-frame/dispatch [:extensions/ethereum-logs nil [nil nil (assoc arguments :address %)]]))))
```

Comment:
Can a avoid an extra dispatch and just provide a reference to a function?  The block in the else clause would be this function.

Reply:
Yes, you're right. Btw the PR was just a placeholder for the gitcoin.